include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -783,12 +786,31 @@ class gap_decorator_anchor_set
     }
 
     //!\brief Stores a (copy of a) view to the ungapped, underlying sequence.
-    decltype(view::all(std::declval<inner_type const &>())) ungapped_view{};
+    decltype(view::all(std::declval<inner_type &&>())) ungapped_view{};
 
     //!\brief Set storing the anchor gaps.
     anchor_set_type anchors{};
 };
 
+/*!\name Type deduction guides
+ * \{
+ */
+//!\brief Ranges (non views!) always deduce to `const & range_type` since they are access-only anyway.
+template <std::ranges::ViewableRange urng_t>
+//!\cond
+    requires std::ranges::RandomAccessRange<urng_t> && std::ranges::SizedRange<urng_t>
+//!\endcond
+gap_decorator_anchor_set(urng_t && range) -> gap_decorator_anchor_set<std::remove_reference_t<urng_t> const &>;
```

Comment:
Does this not mean, that a lvalue view will use this deduction guide? And then you don't want to add const to the type don't you.

Reply:
I don't think so because the type requirements from the deduction guide below are `std::View` which is more constrained than `std::ViewableRange` so this should kick in before overload resolution considers && vs copy no?

Reply:
It works as long as the view overload below does not have &&, which it doesn't. So this is all fine.

Reply:
Yes, we removed that explicitly for that reason

Reply:
Why do you need to comment this?